### DISCUSSION TITLE
How to uninstall the coder's code-server and its autostart

### DISCUSSION THREAD

**Discussion by pochoi** [coder/code-server#5882] [Q&A] [answered]
Hi!

I wanted to install Microsoft code-server, but I accidentally came to 
https://github.com/coder/code-server/ and run the script `curl -fsSL https://code-server.dev/install.sh | sh` and setup the autostart with `systemctl`.

I realize my mistake. I should have not run codes from a publisher I don't trust (yes, I don't trust `coder/code-server` because I think is a sneaky name that tricks people , no matter how many stars it has).

Now I have installed the Microsoft code-server and I am trying to get rid of the coder's code-server.
I follow https://coder.com/docs/code-server/latest/install#uninstall to run
`rm -rf ~/.local/lib/code-server-*`. However, I found that there is still the coder's code-server in my `http://127.0.0.1:8080/login`.

How can I completely get rid coder's coder-server?

My system is Ubuntu 22.04 and `127.0.0.1:8080` code-server has the version
```
code-server: v$VERSION
Code: 1.73.1
Commit: 6261075646f055b99068d3688932416f2346dd3b
Date: 2022-12-02T20:57:48.844Z
Browser: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36
```

I know it is different from my Microsoft code-server because that one has version `Version: 1.75.0-insider`

Thanks!

Po

**Comment by jsjoeio** [comment, marked as answer]
Sorry to hear that! I believe they changed theirs and theirs now comes with the `code` CLI if I'm not mistaken.

> yes, I don't trust coder/code-server because I think is a sneaky name that tricks people , no matter how many stars it has

FWIW, this project was around long before Microsoft came out with their "code-server". 

> http://127.0.0.1:8080/login

Hmm...you should be able to clear your cache, localStorage, etc in your browser settings.

For the `systemctl` side, I found this: https://superuser.com/a/936976/1240803

Hope that helps!

**Reply by lg-nineoneninethreenine** [reply]
> FWIW, this project was around long before Microsoft came out with their "code-server".

Thought of telling this, but you told it bro!

code-server started at 2019(if I am not wrong). I joined this cool community at 2021, May(be?) and have been hanging out since, helping as much as I can with issues, so I know a bit of its history. code-server has been around since 2019 and a lot of YouTuber put this on videos like portable devsystem complete with IDE, or coding on iPads, etc. Microsoft's implementation came just this year, and to be honest, it SUCKS. I tried it on my 4 GB RAM Linux computer and it wouldn't work, unlike code-server, since the requirements are entirely different. I think Microsoft made it for servers while this is made for everyone, so  you can try this out if Microsoft's product seems cumbersome, I'm sure code-server won't let you down(but you need to use some different ways to get your extensions on this since this ain't official).

**Reply by jsjoeio** [reply]
Yeah, you've helped a TON and we super appreciate it!

I do think their implementation is very similar. I think the biggest difference is it can't be self-hosted AFAIK? So if that matters to you, code-server is the best option.

**Reply by lg-nineoneninethreenine** [reply]
TBH Microsoft's code-server is super hard to run on my Intel Core i5 system(like most Microsoft things pre-2019). I haven't tested on other computers, one of them having an ancient Pentium 😂 but we both know it fails on both of them if it fails on my fav Dell Latitude.

I would prefer this, but I edit mostly with Neovim. My laptop keyboard is busted, with circuit errors for F9 and the uparrow key. Vim has hjkl so I like that. But for GUI I would choose this, it's light and better.

**Reply by jsjoeio** [reply]
Oh no way! I'm trying to transition to Neovim. I gave a talk at Neovim Conf 2022. I'll have to link it to you when it's out - it's actually about code-server too!

**Reply by pochoi** [reply]
My bad! So it is actually microsoft repeats the name code-server that causes confusion. Microsoft code server also uses `code-server`. You can find it in https://code.visualstudio.com/blogs/2022/07/07/vscode-server and the binary installed with the script is called `code-server`.

Somehow I nuked my system with other things so I am not in the trouble of distingishing the different code-server anymore. However, I still believe that `rm -rf ~/.local/lib/code-server-*` did not uninstall coder's code server that I installed. As in https://github.com/coder/code-server/discussions/1989 pointing out that coder's code server may be possiby in different locations, the situation becomes harder when you also have Microsoft code server around.

**Reply by lg-nineoneninethreenine** [reply]
Yeah. I tried it on a Zorin install. It messed up due to Python, but I think, Microsoft would have had a small hand on this, perhaps by data paths... I bet it sucks on daily drivers like mine.
